### DISCUSSION TITLE
Nodemcu esp8266 direct connect to s403 port issue

### DISCUSSION THREAD

**Discussion by cuongdm75** [revk/xESP32-Faikout#262] [Q&A]
My unit model is CTXM50RVMV. i used nodemcu esp8266 and firmware from amiga repo. I connected 8266 board directly to my unit PCB through port S403 as in amiga's manual.
Esp booted as expected but it always said: "System is off line."
I would like to attach the screenshot and connection photo. Please help me to figure out what can I do with those.
![20240308_154407](https://github.com/revk/ESP32-Faikin/assets/80936200/d9d2f640-464e-4b3c-bf71-a7169be006e8)

![Screenshot_20240307_213714_Samsung Internet](https://github.com/revk/ESP32-Faikin/assets/80936200/1d2291cb-00f8-41dc-8849-a3676a05201f)

**Comment by revk** [comment]
Not sure if S403 is electrically isolated from mains, so be careful.

No idea if your model will accept 3.3V signalling, if not then it won't work directly connected to a 3.3V GPIO. Some models do work, some don't. This is why my Faikin hardware has a FET.

Make sure you have tx and rx the right way round and configured with the right GPIOs. The code tries polarity inversions by default though. I think the 8266 port may do the same.

**Reply by cuongdm75** [reply]
Ah, I see but I'm not sure my device support 3.3V signal or not. How can I check on my hardware?

**Reply by revk** [reply]
Only by trying 5V and seeing it works, I expect. Someone else may have worked it out. There is a wiki of various models.

**Reply by cuongdm75** [reply]
Thank you for your reply. I will try to use a logic conveter to decrease signal from s403 before esp8266 and will back the result.

**Reply by revk** [reply]
5V in to the ESP8266 is less likely to be an issue, it is most likely the 3.3V to the air-con that needs changing.

**Reply by cuongdm75** [reply]
Thank you for your suggestion, I'll use bi-direction logic converter to go back from 3.3V to 5V of S403. I'm waiting for the parts comming and will be back with result.
